### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by koosoku**

### CODE REVIEW

File: src/controllers/users_controller.js [hunk 1]
```diff
@@ -5,23 +5,29 @@ module.exports.setUp = () => {
 };
 
 module.exports.findAllUsers = (req, res) => {
-    res.json(usersModel.findAllUsers());
+    usersModel.findAllUsers((err, users) => {
+        res.status(err ? 500 : 200).json(users);
```
Comment (silverAndroid) [inline src/controllers/users_controller.js]: If there's an error, why is it sending the array of users down?

File: src/controllers/users_controller.js [hunk 2]
```diff
@@ -5,23 +5,29 @@ module.exports.setUp = () => {
...
+    });
 };
 
 module.exports.register = ({body: {name = null, email = null, password = null}}, res) => {
-    res.json(usersModel.registerUser(name, email, password));
+    if (!email || !password || !name) {
+        return res.status(400).json("Email, name and Password must be provided");
+    }
+    usersModel.registerUser(name, email, password, (err) => {
+        res.status(err ? 500 : 200).json();
```
Comment (silverAndroid) [inline src/controllers/users_controller.js]: Isn't it supposed to return an object here?

File: src/controllers/users_controller.js [hunk 3]
```diff
@@ -5,23 +5,29 @@ module.exports.setUp = () => {
...
-    res.json(usersModel.findAllUsers());
+    usersModel.findAllUsers((err, users) => {
+        res.status(err ? 500 : 200).json(users);
+    });
 };
 
 module.exports.register = ({body: {name = null, email = null, password = null}}, res) => {
-    res.json(usersModel.registerUser(name, email, password));
+    if (!email || !password || !name) {
+        return res.status(400).json("Email, name and Password must be provided");
```
Comment (silverAndroid) [inline src/controllers/users_controller.js]: That's not valid JSON 😛

File: src/controllers/users_controller.js [hunk 4]
```diff
@@ -5,23 +5,29 @@ module.exports.setUp = () => {
...
+    });
 };
 
 module.exports.deleteUser = ({body: {email = null, password = null}}, res) => {
     res.json(usersModel.deleteUser(email, password));
 };
 
 module.exports.login = ({body: {email = null, password = null}}, res) => {
+    if (!email || !password) {
+        return res.status(400).json("Email and Password must be provided");
```
Comment (silverAndroid) [inline src/controllers/users_controller.js]: Invalid JSON

File: src/controllers/users_controller.js [hunk 5]
```diff
@@ -5,23 +5,29 @@ module.exports.setUp = () => {
...
+    }
     usersModel.login(email, password, (err, token) => {
-        if (err) {
-            res.json(err.message);
-        } else {
-            res.json(token);
-        }
+        res.status(err ? 401 : 200).json({token});
```
Comment (silverAndroid) [inline src/controllers/users_controller.js]: Is this by design, returning `{ token: null }` when there's an error?

File: config.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require("dotenv/config");
+
+const config = {};
+
+config.env = process.env.NODE_ENV || "production";
```
Comment (silverAndroid) [inline config.js]: Why is this not inside the config object?

File: src/models/recipes_model.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 const cassandra = require("cassandra-driver");
 const async = require("async");
 const client = new cassandra.Client({contactPoints: ["127.0.0.1"]});
-const environment = "development";
-
+const config = require("../../config");
+const environment = config.env;
```
Comment (silverAndroid) [inline src/models/recipes_model.js]: Inconsistent with users_model line 6

File: src/models/users_model.js [hunk 8]
```diff
@@ -113,7 +160,7 @@ module.exports.deleteUser = (email, password) => {
         (next) => fetchUserInfo(email, next),
         (userInfo, next) => auth.authorizeLogin(email, password, userInfo, next),
         (userInfo, next) => deleteUser(userInfo, next)
-    ], onResultReturned);
+    ]);
```
Comment (silverAndroid) [inline src/models/users_model.js]: Is this supposed to have `callback` as well?

File: src/routes/users_routes.js [hunk 9]
```diff
@@ -1,7 +1,10 @@
 const app = require("express").Router();
 const usersController = require("../controllers/users_controller");
+const config = require("../../config");
```
Comment (silverAndroid) [inline src/routes/users_routes.js]: You can destructure the config object so you just have env

File: src/controllers/users_controller.js (hunk 2 above)
Comment (koosoku) [reply src/controllers/users_controller.js]: status should suffice

File: src/controllers/users_controller.js (hunk 1 above)
Comment (koosoku) [reply src/controllers/users_controller.js]: i'll fix this :)

File: src/controllers/users_controller.js (hunk 3 above)
Comment (koosoku) [reply src/controllers/users_controller.js]: oh true...

File: src/controllers/users_controller.js (hunk 4 above)
Comment (koosoku) [reply src/controllers/users_controller.js]: true

File: src/controllers/users_controller.js (hunk 5 above)
Comment (koosoku) [reply src/controllers/users_controller.js]: okay, i'll return nothing

File: config.js (hunk 6 above)
Comment (koosoku) [reply config.js]: will move

File: src/routes/users_routes.js (hunk 9 above)
Comment (koosoku) [reply src/routes/users_routes.js]: true

**Pull request merged**